tests/marine/test_marine.py

```diff
@@ -1025,3 +1025,34 @@ def test_parse_fields_preserves_order(marine_instance: Marine, tcp_packet: bytes
         "udp.srcport": None,
         "tcp.srcport": "16424",
     }
+
+
+def test_parse_all_fields_int_value(tcp_packet_fields):
+    tcp_source_port = tcp_packet_fields["tcp"]["tcp.srcport"]
+    assert isinstance(tcp_source_port, int)
+    assert tcp_source_port == 16424
+
+
+def test_parse_all_fields_str_value(tcp_packet_fields):
+    ip_src = tcp_packet_fields["ip"]["ip.src"]
+    assert isinstance(ip_src, str)
+    assert ip_src == "10.0.0.255"
+
+
+def test_parse_all_fields_list_value(tcp_packet_fields):
+    ip_addr = tcp_packet_fields["ip"]["ip.addr"]
+    assert isinstance(ip_addr, list)
+    assert "10.0.0.255" in ip_addr
+    assert "21.53.78.255" in ip_addr
+
+
+def test_parse_all_fields_bool_value(tcp_packet_fields):
+    tcp_ack_flag = tcp_packet_fields["tcp"]["tcp.flags_tree"]["tcp.flags.ack"]
+    assert isinstance(tcp_ack_flag, bool)
+    assert tcp_ack_flag
```

Comment:
For safety, maybe add another assert here for a flag that is 0?

Reply:
As in - the bit extracted from the dissector is 0? 
Doesn't isn't the assert on line 1052 enough? it would fail if the flag is `False`.

Reply:
I meant checking that a different tcp flag is 0